Integrate Flowbite-Svelte-Icons

shinokada opened the pull request:
Closes #893

## Status

- [ ] Not Completed
- [x] Completed

## ✅ Checks

<!-- Make sure your PR passes the tests and do check the following fields as needed - -->

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [ ] I have checked the page with https://validator.unl.edu/
- [x] All the tests have passed
- [x] My pull request is based on the latest commit (not the npm version).
### Describe the bug

Flowbite has SVG icon updates, https://github.com/themesberg/flowbite/pull/588.

- We'd like to update [Flowbtite-Svelte](https://flowbite-svelte.com/) using  [Flowbite-Svelte-Icons](https://github.com/themesberg/flowbite-svelte-icons)
- Flowbite-Svelte-Icons will be in the devDependencies but not in the peerDependencies.

## Docs pages

- [x] Accordion
- [x] Alerts (InfoCircleSolid, CloseOutline, EyeSolid)
- [x] Badge (ClockSolid, EnvelopSolid, CheckOutline)
- [x] Banner
- [x] Bottom nav
- [x] Breadcrumb (*improved content and spacing for advanced examples)
- [x] Button group
- [x] Button (*added new button icon sizes)
- [x] Card
- [x] Drawer
- [x] Dropdowns
- [x] Footer (*improved colors and spacing)
- [x] Forms
- [x] Indicator
- [x] List group
- [x] Mega menu
- [x] Modal
- [x] Navbar
- [x] Pagination (*added new sizes and fixed paddings)
- [x] Popover (*updated icon & added dropdowns)
- [x] Rating (* improve colors and spacing)
- [x] Sidebar (we need the sidebar menu toggle icon @robert1508)
- [x] Speed Dial
- [x] Tabs
- [x] Timeline
- [x] Toast
- [x] Checkbox
- [x] File input
- [x] Input field
- [x] Radio
- [x] Search input
- [x] Select
- [x] Textarea
- [x] Blockquote
- [x] Headings
- [x] HR
- [x] Links
- [x] Lists
- [x] Paragraphs
- [x] Text decoration

jjagielka commented on src/routes/docs/components/alert.md:
```diff
@@ -67,29 +67,30 @@ You can also include a descriptive icon to complement the message inside the ale
 </script>
 <Alert>
-  <svg slot="icon" aria-hidden="true" class="w-5 h-5" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z" clip-rule="evenodd"></path></svg>
+  <InfoCircleSolid  class="flex-shrink-0 text-primary-800 dark:text-primary-400 inline w-4 h-4 mr-3" />
```
Why don't you use `slot="icon"` here? If you did the only settings needed would be `size="sm"` or `class="w-4 h-4"`

shinokada replied:
Thank you for the catch. I update them.

jjagielka commented on src/routes/docs/components/badge.md:
```diff
@@ -121,13 +121,14 @@ You can also use SVG icons inside the badge elements.
 </script>
 <Badge color="dark" border>
-  <svg aria-hidden="true" class="w-3 h-3 mr-1" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M10 18a8 8 0 100-16 8 8 0 000 16zm1-12a1 1 0 10-2 0v4a1 1 0 00.293.707l2.828 2.829a1 1 0 101.415-1.415L11 9.586V6z" clip-rule="evenodd"></path></svg>
+  <ClockSolid class="text-gray-800 dark:text-gray-400 w-2.5 h-2.5 mr-1.5"/>
```
Setting `text-gray-800 dark:text-gray-400` is redundant here as those are inherited from the surrounding badge.

shinokada replied:
Thanks. Updated.

jjagielka commented on src/routes/docs/components/card.md:
```diff
@@ -306,32 +310,32 @@ Show detailed information to potential customers about your product’s pricing
       <li class="flex space-x-2">
-            <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="w-6 h-6 text-primary-600 dark:text-primary-500"><path stroke-linecap="round" stroke-linejoin="round" d="M9 12.75L11.25 15 15 9.75M21 12a9 9 0 11-18 0 9 9 0 0118 0z" /></svg>
-          <span class="text-base font-normal leading-tight text-gray-500 dark:text-gray-400">Integration help</span>
+        <CheckCircleSolid class="flex-shrink-0 w-4 h-4 text-primary-600 dark:text-primary-500" />
```
I think `class="flex-shrink-0"` should be added to all icons by default. You can move that to `flowbite-svelte-icons`.

shinokada replied:
Added to flowbite-svelt-iconsv0.2.4. Can you check it, please?

shinokada replied:
Updated.

jjagielka commented on src/routes/docs/components/forms.md:
```diff
@@ -83,29 +83,34 @@ With the Input component, you can add <a href="https://flowbite-svelte.vercel.ap
-  <svg slot="left" aria-hidden="true" class="w-5 h-5 text-gray-500 dark:text-gray-400" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M2.003 5.884L10 9.882l7.997-3.998A2 2 0 0016 4H4a2 2 0 00-1.997 1.884z"></path><path d="M18 8.118l-8 4-8-4V14a2 2 0 002 2h12a2 2 0 002-2V8.118z"></path></svg>
-  </Input>
+  <ButtonGroup class="w-full">
+    <InputAddon id="email" type="email" placeholder="name@flowbite.com">
```
This is a different example. It's not the `InputAddon` here but the floating icon. So `slot="left"` should be used at `Envelope`.

shinokada replied:
Thanks. Updated.

jjagielka commented on src/routes/docs/components/alert.md:
```diff
@@ -105,27 +105,27 @@ Use this example to add a border accent to the alert component instead of just a
 </script>
 <Alert border>
-  <InfoCircleSolid  class="flex-shrink-0 text-primary-800 dark:text-primary-400 inline w-4 h-4 mr-3" />
+  <InfoCircleSolid  class="text-primary-800 dark:text-primary-400 inline w-4 h-4 mr-3" />
```
You've corrected to `slot="icon"` in the first icon example, but it's valid for all alerts with icons: bordered, dismissable, etc

shinokada replied:
Updated.

shinokada commented:
Now it is ready to be merged. Please check it and let me know.

- Replaced all svg icons with flowbite-svelte-icons
- Moved svg icons that are not in flowbite-svelte-icons to utils dir.
- Replaced Chevron, ChevronUp, ChevronDown, ChevronRight, and ChevronLeft with flowbite-svelte-icons
- Removed and cleaned unnecessary files.
- Finally, I can run `pnpm format` for markdown files as well.

jjagielka commented on src/lib/typography/Hr.svelte:
```diff
@@ -2,19 +2,21 @@
 </script>
 
-{#if $$slots}
+{console.log($$slots.default)}
```
Left over. To remove.

shinokada replied:
Thanks. Updated.

jjagielka commented on src/routes/docs/+layout.svelte:
```diff
@@ -64,8 +62,34 @@
+                stroke-linejoin="round"
+                stroke-width="2"
+                d="m1 1 4 4 4-4" />
+            </svg>
```
This is `+layout.svelte` from docs. This is not a library component. You're free to use `<ChevronDownOutline/>` here.

shinokada replied:
Thanks. Updated.

jjagielka commented on src/routes/docs/components/banner.md:
```diff
@@ -55,93 +59,121 @@ Use this free example to show a text message for announcement with a CTA link, a
+      type="submit"
+      class="text-white bg-primary-700 hover:bg-primary-800 focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm w-full sm:w-auto px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
+      >Subscribe</button>
+  </form>
```
This formatting (wrapping long lines) is discourage in the examples. It decrease the readability. See:
```svelte
 <label
      for="email"
      class="flex-shrink-0 mb-2 mr-auto text-sm font-medium text-gray-500 md:mb-0 md:mr-4 dark:text-gray-400 md:m-0"
      >Sign up for our newsletter</label>
    <input
      type="email"
      id="email"
      placeholder="Enter your email"
      class="bg-white border border-gray-300 text-gray-900 md:w-64 mb-2 md:mb-0 md:mr-4 text-sm rounded-lg focus:ring-primary-500 focus:border-primary-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white dark:focus:ring-primary-500 dark:focus:border-primary-500"
      required />
    <button
      type="submit"
      class="text-white bg-primary-700 hover:bg-primary-800 focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm w-full sm:w-auto px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
      >Subscribe</button>
```
vs
```svelte
   <label for="email"  class="flex-shrink-0 mb-2 mr-auto text-sm font-medium text-gray-500 md:mb-0 md:mr-4 dark:text-gray-400 md:m-0">
     Sign up for our newsletter
    </label>
    <input  type="email"  id="email" placeholder="Enter your email" class="bg-white border border-gray-300 text-gray-900 md:w-64 mb-2 md:mb-0 md:mr-4 text-sm rounded-lg focus:ring-primary-500 focus:border-primary-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white dark:focus:ring-primary-500 dark:focus:border-primary-500"  required />
    <button type="submit" class="text-white bg-primary-700 hover:bg-primary-800 focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm w-full sm:w-auto px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800">
       Subscribe
   </button>
```

jjagielka replied:
Unfortunately it's in lots of places.

shinokada replied:
We need to update .prettierrc config:

```
{
  "useTabs": false,
  "singleQuote": true,
  "trailingComma": "none",
  "printWidth": 109,
  "embeddedLanguageFormatting": "auto",
  "bracketSameLine": true
}
```

shinokada replied:
I think increasing the `printWidth` value may work.

jjagielka replied:
Yes, but that must be set for markdown files only. In the normal code wrapping at 109 chars is a good thing.

jjagielka commented:
> Finally, I can run pnpm format for markdown files as well.

This can do (and already did in lot's of places) more harm then good. The readability of the examples is worsen with wrapping long lines. Should we set markdown formatting line length to huge value like 1024?

shinokada commented:
https://prettier.io/docs/en/options.html#prose-wrap this may work.

jjagielka commented:
Yes, that may work. Value `never` should be used?

shinokada commented:
I will revert the commit and will try it.

jjagielka commented:
From the `props-wrap` docs:
> "never" - Un-wrap each block of prose into one line.

Maybe you don't need to revert, but reapply the formatting and lines would be `un-wrapped`?

shinokada commented:
I tried different things but ` "printWidth": 1024,` is the best one.

shinokada commented:
@zoltanszogyenyi Please let me know if I can merge this PR.

zoltanszogyenyi commented:
@shinokada I'm currently traveling, will check in a couple of hours! Thanks.

zoltanszogyenyi commented:
Hey @shinokada @jjagielka,

First of all - great job so far on integrating the new icons!

Here's a list of feedback from my part based on what I saw so far:

- the PR is not yet ready for merge because some icons are still either too large or small or the spacing is a bit off, for example for the banner components:

<img width="959" alt="Screenshot 2023-07-13 at 11 26 44" src="https://github.com/themesberg/flowbite-svelte/assets/8052108/22ed46c4-9780-4bc2-9d39-d6a36f3bdac3">
<img width="935" alt="Screenshot 2023-07-13 at 11 26 28" src="https://github.com/themesberg/flowbite-svelte/assets/8052108/4ddbff73-1f22-45a4-a336-f336509afa16">

Because we added all the new icons for the main Flowbite library, you can copy over the sizes and spaces directly from there for consistency: https://flowbite.com/blocks/marketing/banner/

- I suggest that for this same PR we also integrate the new icons into the website and docs itself, for example, change the icons for the navbar, for the code preview, dark mode icon, etc just the same way we did for the main library from which we can take these examples

I'll have time again to take a look on Monday - currently, I'm on leave from my home town.

But super great job so far, we're definitely much closer to launching this now!

Cheers,
Zoltan

shinokada commented:
@zoltanszogyenyi Ok, I need help here. Can you list which pages and icons are not the same as flowbite.com?

shinokada commented:
@jjagielka @zoltanszogyenyi 
I just updated to the new version of flowbite-svelte-icons and went through the details. Please let me know if I missed anything. Please be specific what I need to fix.

shinokada commented:
Since there is no response, I will merge this PR.

shinokada commented:
No reason.
I just copied from https://flowbite.com/ and it was easier for me.
